pkg/services/ngalert/api/api_ruler_validation.go

```diff
@@ -154,20 +188,34 @@ func validateLabels(l map[string]string) error {
 	return nil
 }
 
-func validateCondition(condition string, queries []apimodels.AlertQuery) error {
+func validateCondition(condition string, queries []apimodels.AlertQuery, canPatch bool) error {
+	if canPatch {
+		// Patch requests may leave both query and condition blank. If a request supplies one, it must supply the other.
+		if len(queries) == 0 && condition == "" {
+			return nil
+		}
+		if len(queries) == 0 && condition != "" {
+			return fmt.Errorf("%w: query is not specified but condition is. You must specify both query and condition to update existing alert rule", ngmodels.ErrAlertRuleFailedValidation)
+		}
+		if len(queries) > 0 && condition == "" {
+			return fmt.Errorf("%w: condition is not specified but query is. You must specify both query and condition to update existing alert rule", ngmodels.ErrAlertRuleFailedValidation)
+		}
+	}
```

Comment:
This is a less nested version of the if/else logic surrounding validateCondition's old call site. IMO it shows intent much better.

Reply:
Agreed, much easier to follow 👍